### PR TITLE
Guest user

### PR DESCRIPTION
**Pull request by bjoernricks**
Support for guest user login. With this change a config.js file with the following content

```javascript
config = {
    guestUsername: 'foo',
    guestPassword: 'bar',
}
```
can be provided to activate a `Login as Guest` button at the login page. To actually use this user
a user with the role Guest and the specified username/password combination must be created.

**Important**: It is not checked if the user is a restricted guest user. It would also be possible to auto login via a super admin user!

Additionally by adding a `disableLoginForm: true` to the config object in the `config.js` file it is possible to deactivate the *normal* login form at the login page to only provide the `Login as Guest` button.

**Checklist**:

- [x] Tests
- [x] [CHANGES](https://github.com/greenbone/gsa/blob/master/CHANGES.md) Entry

### CODE REVIEW

File: gsa/src/web/pages/login/loginform.js [hunk 1]
```diff
@@ -93,62 +101,85 @@ class LoginForm extends React.Component {
   }
 
   render() {
-    const {error} = this.props;
+    const {
+      error,
+      showGuestLogin = false,
+      showLogin = false,
```
Comment (swaterkamp) [inline gsa/src/web/pages/login/loginform.js]: If this is false by default, shouldn't NOT setting it to true explicitly on the LoginPage not show the LoginForm? I.e. when rendering ```<LoginForm/>``` we would need the prop ```showLogin={true}```

Comment (swaterkamp) [reply gsa/src/web/pages/login/loginform.js]: Nevermind. It's done in the next commit...

File: gsa/src/web/utils/testing.js [hunk 2]
```diff
@@ -52,6 +52,14 @@ afterEach(cleanup);
 const queryAllByName = (container, name) =>
   queryAllByAttribute('name', container, name);
 
+const queryByName = (container, name) => {
+  const els = queryAllByName(container, name);
```
Comment (swaterkamp) [inline gsa/src/web/utils/testing.js]: What does els stand for? Without the proper background knowledge this is not a descriptive variable name for me

Comment (bjoernricks) [reply gsa/src/web/utils/testing.js]: It uses the same name pattern as dom-testing-library:

queryAll... => return all found elements as array or empty array if no element is found
query... => return first found element or null if no element is found
get... => return first found elment or raise exception if no element is found
...ByTestId => search by data-test-id attribute
...ByName => search by name attribute

File: gsa/src/web/pages/login/loginform.js (hunk 1 above)
Comment (bjoernricks) [reply gsa/src/web/pages/login/loginform.js]: It's updated to showLogin=true in a later commit ;-)

File: gsa/src/web/utils/testing.js (hunk 2 above)
Comment (swaterkamp) [reply gsa/src/web/utils/testing.js]: Thanks for the comprehensive answer. My question was merely about the variable naming though ;)

Comment (bjoernricks) [reply gsa/src/web/utils/testing.js]: `els` is an abbreviation for elements. I've used the same variables as in dom-testing-library
https://github.com/kentcdodds/dom-testing-library/blob/master/src/queries.js#L225

**Pull request merged**